docs/targeter.md

```diff
@@ -0,0 +1,67 @@
+
+* `drain`: How much power this Targeter will take.
+* `attackSpeed`: How long between attacks. Given in milliseconds, eg 500ms means it will attack twice per second.
+* `range`: The radius of this Targeter's range. Given in blocks.
+
+Additionally there is an abstract method called `getMultiplier` that the component will need to implement. This method returns a float that is passed to the [Effectors](effector.md) each attack. It should be used as a balancing tool to make sure that the different targeters are all correctly balanced.
+
+Of particular note is the `SingleTargeter` as this is a base class for all bar one implementations. It has an additional field called `selectionMethod`. This is a value from the [`SelectionMethod` enum](http://jenkins.terasology.org/view/Modules/job/GooeyDefence/javadoc/org/terasology/gooeyDefence/towers/SelectionMethod.html) that provides the player with control over how the targeter selects its targets.
+
+Here example of what a targeter called `HeightTargeter` that added a float field `minHeight`, and extended SingleTargeter might look:
+
+    public class SingleTargeterComponent extends SingleTargeterComponent {
+        public float minHeight = 0.5f;
+        @Override
+        public float getMultiplier() {
+            return 1.2f;
+        }
+    }
+
+### System
+
+The second half of this is the system. This will implement the actual selection of enemies using the component detailed in the prior section. Unlike the component, this system does not need to extend any special classes and can be a plain system. It can even have other responsibilities and functionality just like all systems. It is recommended to extend the [`BaseTargeterSystem`](http://jenkins.terasology.org/view/Modules/job/GooeyDefence/javadoc/org/terasology/gooeyDefence/towers/targeters/BaseTargeterSystem.html) however, for reasons explained below. When a targeter is called to select its targets, the [`SelectEnemiesEvent`](http://jenkins.terasology.org/view/Modules/job/GooeyDefence/javadoc/org/terasology/gooeyDefence/towers/events/SelectEnemiesEvent.html) is sent. The event handler should simply use the `addToList` methods to add enemies to the event. Managers such as [`EnemyManager`](http://jenkins.terasology.org/view/Modules/job/GooeyDefence/javadoc/org/terasology/gooeyDefence/EnemyManager.html) have some useful methods which will help in selecting enemies. Additionally the system, `BaseTargeterSystem`, has a number of useful helper methods which can simplify the process.
+Any visual effects should also be triggered in this event, helper methods for which are located in [`InWorldRenderer`](http://jenkins.terasology.org/view/Modules/job/GooeyDefence/javadoc/org/terasology/gooeyDefence/visuals/InWorldRenderer.html).
+
+As an example here is the entirety of `AoeTargeterSystem`, minus javadoc, which serves as one of the simpler implementations:
+
+    @RegisterSystem
```

Comment:
Does this need to be wrapped in code syntax markup? That might be another thing that could get lost via copy paste. Usually I just stick three tildes on their own lines before and after :-)

I wish there was an easier way to preview markdown files right in a GitHub PR 😅 

(This is just one spot there appear to be multiple)

Reply:
Markdown only requires a code block to be indented actually, so it shows up as code when I preview it. I could change it to tildes if that would be clearer?

Reply:
Oh that's fine then. That's on me not having the chance to preview :-)